plugin/core/sessions.py

```diff
@@ -1704,9 +1708,9 @@ def notify_plugin_on_session_buffer_change(self, session_buffer: SessionBufferPr
         if self._plugin:
             self._plugin.on_session_buffer_changed_async(session_buffer)
 
-    def _maybe_resolve_code_action(self, code_action: CodeAction) -> Promise[Union[CodeAction, Error]]:
-        if "edit" not in code_action and self.has_capability("codeActionProvider.resolveProvider"):
-            # TODO: Should we accept a SessionBuffer? What if this capability is registered with a documentSelector?
+    def _maybe_resolve_code_action(self, code_action: CodeAction, session_buffer: SessionBufferProtocol) -> Promise[Union[CodeAction, Error]]:
+        has_capability =  self.has_capability("codeActionProvider.resolveProvider") or session_buffer.has_capability("codeActionProvider.resolveProvider")
```

Comment:
`SessionBuffer.has_capability` falls back to the capabilities of the session, so I think using only the second part here should be sufficient. The next line has two spaces after `and` and you could move this condition directly to the next line instead of defining a variable for it. But I would probably pass the view and defer getting the SessionBuffer to do it only when necessary, like this:
```python
if "edit" not in code_action:
    session_buffer = self.get_session_buffer_by_id(view.buffer_id())
    if session_buffer and session_buffer.has_capability("..."):
        request = ...
```